src/main/java/com/assignment/application/controller/SalaryController.java

```diff
@@ -14,30 +18,93 @@
 public class SalaryController {
 
     @Autowired
-    SalaryServiceI salaryServiceI;
+    private SalaryServiceI salaryServiceI;
 
-    @PostMapping(value="{comp_id}/{emp_id}/salary" )
+    @Autowired
+    private KafkaTemplate<String, SalaryUpdate> kafkaTemplateSalary;
+
+    @Autowired
+    private VerifyUser verifyUser;
+
+
+    public final String TOPIC = "SalaryUpdate";
+
+
+    @PostMapping(value = "{comp_id}/{emp_id}/salary")
     public ResponseEntity<Salary> addSalaryInfo(@PathVariable("comp_id") Long companyId,
                                                 @PathVariable("emp_id") String employeeId,
-                                                @RequestBody Salary salary){
-        return salaryServiceI.addSalary(companyId,employeeId,salary);
+                                                @RequestBody Salary salary,
+                                                @RequestHeader("username") String username,
+                                                @RequestHeader("password") String password) {
+        try {
+            if (verifyUser.authorizeUser(username, password) == 1 || (verifyUser.authorizeEmployee(username, password) == 1 && employeeId.equalsIgnoreCase(username))) {
+                Salary salaryToBeAdded = salaryServiceI.addSalary(companyId, employeeId, salary);
+                if (salary == null) {
+                    return new ResponseEntity<>(null, HttpStatus.BAD_REQUEST);
+                }
+                return new ResponseEntity<>(salaryToBeAdded, HttpStatus.OK);
+            }
+            return new ResponseEntity<>(null, HttpStatus.UNAUTHORIZED);
+        } catch (Exception e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(null, HttpStatus.BAD_REQUEST);
+        }
     }
 
-    @GetMapping(value="{comp_id}/{emp_id}/salary" )
+    @GetMapping(value = "{comp_id}/{emp_id}/salary")
     public ResponseEntity<Salary> getSalaryInfo(@PathVariable("comp_id") Long companyId,
-                                                @PathVariable("emp_id") String empId){
-        return salaryServiceI.getSalary(companyId,empId);
+                                                @PathVariable("emp_id") String employeeId,
+                                                @RequestHeader("username") String username,
+                                                @RequestHeader("password") String password) {
+        try {
+            if (verifyUser.authorizeUser(username, password) == 1 || (verifyUser.authorizeEmployee(username, password) == 1 && employeeId.equalsIgnoreCase(username))) {
+                Salary salary = salaryServiceI.getSalary(companyId, employeeId);
+                if (salary == null) {
```

Comment:
This is not a bad request scenario. It is possible in correct cases as well. Just the salary info is yet to be uploaded.